DevCycle.SDK.Server.Local.MSTests/EventQueueTest.cs

```diff
@@ -33,15 +34,254 @@ public class EventQueueTest
             return new Tuple<EventQueue, MockHttpMessageHandler, MockedRequest>(eventQueue, mockHttp, req);
         }
 
+        [TestMethod]
+        public async Task TestQueueLimit_OneUserAggregate()
+        {
+            var localOptions = new DVCLocalOptions(1000, 1000);
+            var eventQueue = getTestQueue(
+                localOptions: localOptions);
+            var user = new User("user1");
+            eventQueue.Item1.AddFlushedEventsSubscriber((_, args) =>
+            {
+                Console.WriteLine("Error?: " + args.Error);
+                Console.WriteLine("Flushed events: " + (args.Success ? "true" : "false"));
+            });
+            var loopsCompleted = 0;
+
+            Assert.ThrowsException<DVCException>(() =>
+            {
+                for (int i = 0; i < 10000; i++)
+                {
+                    eventQueue.Item1.QueueAggregateEvent(new DVCPopulatedUser(user),
+                        new Event("testEvent" + i, "" + i,
+                            metaData: new Dictionary<string, object> {{"test", "value"}}),
+                        new BucketedUserConfig()
+                        {
+                            FeatureVariationMap = new Dictionary<string, string>
+                                {{"some-feature-id", "some-variation-id"}}
+                        });
+                    loopsCompleted = i;
+                }
+
+                return loopsCompleted;
+            });
+
+            await Task.Delay(5000);
+            var matches = eventQueue.Item2.GetMatchCount(eventQueue.Item3);
+            Console.WriteLine(matches + " matches");
+            Assert.AreEqual(localOptions.MaxEventsInQueue, loopsCompleted);
+        }
+
+
+        [TestMethod]
+        public async Task TestDisableEvents()
+        {
+            var localOptions = new DVCLocalOptions(1000, 1000, disableEvents: true);
+            var eventQueue = getTestQueue(
+                localOptions: localOptions);
+            var user = new User("user1");
+            eventQueue.Item1.AddFlushedEventsSubscriber((_, args) =>
+            {
+                Console.WriteLine("Error?: " + args.Error);
+                Console.WriteLine("Flushed events: " + (args.Success ? "true" : "false"));
+            });
+            eventQueue.Item1.QueueAggregateEvent(new DVCPopulatedUser(user),
+                new Event("testEvent", "target", metaData: new Dictionary<string, object> {{"test", "value"}}),
+                new BucketedUserConfig()
+                {
+                    FeatureVariationMap = new Dictionary<string, string>
+                        {{"some-feature-id", "some-variation-id"}}
+                });
+
+            await Task.Delay(5000);
+            var matches = eventQueue.Item2.GetMatchCount(eventQueue.Item3);
+            Assert.AreEqual(0, matches);
+        }
+
+        [TestMethod]
+        public async Task TestQueueLimit_MultiUserAggregate()
```

Comment:
any way to add some description to these test cases to describe what's going on a bit better?

Reply:
yep - can add